pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -382,14 +382,25 @@ def _resolve_sampler(self, dataloader: DataLoader, shuffle: bool, mode: Optional
                     " distributed training. Either remove the sampler from your DataLoader or set"
                     " `replace_sampler_ddp=False` if you want to use your custom sampler."
                 )
-            return self._get_distributed_sampler(
+            sampler = self._get_distributed_sampler(
                 dataloader,
                 shuffle,
                 mode=mode,
                 overfit_batches=self.trainer.overfit_batches,
                 **self.trainer.distributed_sampler_kwargs,
             )
 
+            trainer_fn = self.trainer.state.fn
+            if isinstance(sampler, DistributedSampler) and trainer_fn in (TrainerFn.VALIDATING, TrainerFn.TESTING):
+                rank_zero_warn(
+                    f"Using `DistributedSampler` with the dataloaders. During trainer.{trainer_fn.value}(),"
+                    " it is recommended to use single device strategy to ensure each sample/batch gets evaluated"
```

Comment:
Let's make it simple for the users

```suggestion
                    " it is recommended to use `devices=1` to ensure each sample/batch gets evaluated"
```